src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Automation/VSImports.cs

```diff
@@ -0,0 +1,194 @@
+            {
+                using (var access = await _lockService.ReadLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    var importsList = project.GetItems(importItemTypeName)
+                                             .Select(p => p.EvaluatedInclude)
+                                             .ToList();
+                    return importsList.ElementAt(lIndex - 1);
+                }
+            });
+        }
+
+        public void Add(string bstrImport)
+        {
+            _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.WriteLockAsync())
+                {
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    await access.CheckoutAsync(project.Xml.ContainingProject.FullPath).ConfigureAwait(true);
+                    project.AddItem(importItemTypeName, bstrImport);
+                }
+            });
+
+            OnImportAdded(bstrImport);
+        }
+
+        public void Remove(object index)
+        {
+            string importRemoved = null;
+            _threadingService.ExecuteSynchronously(async () =>
+            {
+                using (var access = await _lockService.WriteLockAsync())
+                {
+                    Microsoft.Build.Evaluation.ProjectItem importProjectItem = null;
+                    var project = await access.GetProjectAsync(ConfiguredProject).ConfigureAwait(true);
+                    await access.CheckoutAsync(project.Xml.ContainingProject.FullPath).ConfigureAwait(true);
+                    if (index is int indexInt)
+                    {
+                        importProjectItem = project.GetItems(importItemTypeName)
+                                                   .ElementAt((indexInt - 1));
+                    }
+                    else if (index is string removeImport)
+                    {
+                        importProjectItem = project.GetItems(importItemTypeName)
+                                                   .First(i => string.Compare(removeImport, i.EvaluatedInclude, StringComparison.OrdinalIgnoreCase) == 0);
+                    }
+                    else
+                    {
+                        System.Diagnostics.Debug.Assert(false, $"Parameter {nameof(index)} is niether an int nor a string");
+                        return;
+                    }
+
+                    if (importProjectItem.IsImported)
+                    {
+                        throw new ArgumentException(string.Format(VSPackage.ImportParamNotFound, index.ToString()), nameof(index));
+                    }
+
+                    importRemoved = importProjectItem.EvaluatedInclude;
+                    project.RemoveItem(importProjectItem);
+                }
+            });
+
+            OnImportRemoved(importRemoved);
```

Comment:
Don't raise this if it doesn't exist, after you fix the case where it doesn't exist.